### PR TITLE
V2 audit recheck

### PR DESCRIPTION
**Pull request by frontier159**
# Description
What does this PR solve?

Fixes # (issue)

# Checklist
- [ ] Code follows the style guide
- [ ] I have performed a self-review of my own code
- [ ] New and existing tests pass locally
- [ ] This PR is targeting the correct branch

### CODE REVIEW

File: protocol/contracts/v2/TempleDebtToken.sol [hunk 1]
```diff
@@ -664,6 +661,9 @@ contract TempleDebtToken is ITempleDebtToken, TempleElevatedAccess {
 
         /// @dev The risk premium interest rate for this debtor
         uint96 riskPremiumRate;
+
+        /// @dev The total balance this debtor owes: principal + baseInterest + riskPremiumInterest
+        uint128 totalBalance;
```
Comment (princetonbishop) [inline protocol/contracts/v2/TempleDebtToken.sol:666]: should also update `debtorCache.totalBalance` in `_burn` for completeness

Comment (frontier159) [reply protocol/contracts/v2/TempleDebtToken.sol:666]: I considered it, but it wasn't being used by anything downstream so seemed unnecessary

Comment (frontier159) [reply protocol/contracts/v2/TempleDebtToken.sol:666]: Happy to take your opinion on it though?

Comment (princetonbishop) [reply protocol/contracts/v2/TempleDebtToken.sol:666]: I thought the same, so tagged it as nit. let's save gas then. keep it as it is.

**Review by mirionic — APPROVED** [review]

**Pull request merged**